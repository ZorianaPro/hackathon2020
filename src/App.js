import React, { Component } from 'react';

import './App.css';

import { BrowserRouter as Router, Route, Link } from "react-router-dom";

import Books from "./components/Books";

class App extends Component {
  render() {
    return (
      <Router>
        <Route path="/" exact component={Books} />
      </Router>
    );
  }
}

export default App;