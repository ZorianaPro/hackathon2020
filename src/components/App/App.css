.App {
  font-family: 'Lato';
  font-style: normal;
  font-weight: 100;
  color: var(--eerie-black-color);
  font-size: 1em;
  letter-spacing: 0.125em;
}

.App .App-Main-Section-Container {
  margin: 0 auto;
  width: 100%;
  max-width: 1200px;
  height: 100%;
  position: relative;
}

.App .App-Main-Section {
  height: 100vh;
}

.App .App-Main-Section-Title {
  font-size: 3em;
  width: 70%;
  margin: 1em 0;
}

.App .App-Main-Section-Info {
  position: absolute;
  top: 50%;
  left: 0;
  transform: translateY(-50%);
}

.App .App-Main-Section-Info svg {
  width: 2em;
  height: 2em;
}

.App .App-Main-Section-Info p {
  margin: 0.5em;
  font-size: 1.5em;
}

.App .App-Main-Section-Info .App-Main-Section-Date,
.App .App-Main-Section-Info .App-Main-Section-Location {
  display: flex;
  flex-direction: row;
  justify-content: flex-start;
  align-items: center;
}

.App .App-Main-Section-Info .Button {
  margin: 2em 0;
}