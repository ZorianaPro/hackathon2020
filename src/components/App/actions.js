export const actions = {
  fetchAllIdeas: 'FETCH:ALL:IDEAS',
  result: 'RESULT:IDEAS',
  error: 'ERROR:IDEAS'
};

export const fetchAllIdeas = (credentials = '') => ({
  type: actions.fetchAllIdeas,
  value: credentials
});

export const result = (ideas = []) => ({
  type: actions.result,
  value: ideas
});

export const error = (err) => ({
  type: actions.error,
  value: err
});

export default {
  fetchAllIdeas,
  result,
  error
};
