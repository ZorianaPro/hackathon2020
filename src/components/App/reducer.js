const initialState = {
  version: process.env.REACT_APP_VERSION,
};

export default (state = initialState, action) => {
  switch (action.type) {
    default:
      return state;
  }
};
