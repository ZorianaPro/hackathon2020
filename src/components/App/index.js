import React, {useState, useEffect, useCallback} from 'react';
import './App.css';
import Header from "../Header";
import SVGSpriteSheet from '../SVGSpriteSheet';
import LocationSVG from '../LocationSVG';
import DateSVG from "../DateSVG";
import Button from "../Button";
import Ideas from "../Ideas";
import Schedule from '../Schedule';
import Rules from "../Rules";
import Footer from "../Footer";
import Registration from "../Registration";

const App = ({

}) => {

  return (
    <div className="App">
      <Header />
      <section id="about" className="App-Main-Section">
        <div className="App-Main-Section-Container">
          <div className="App-Main-Section-Info">
            <div className="App-Main-Section-Title">
              Cool name for our cool hackathon
           </div>
            <div className="App-Main-Section-Date">
              <DateSVG />
              <p> 12-15 December 2020 </p>
            </div>
            <div className="App-Main-Section-Location">
              <LocationSVG />
              <p> Online </p>
            </div>
            <Button title='Go to registration'
              action='scroll'
              target='Registration' />
          </div>
        </div>
      </section>
      <hr />
      <section id="schedule" className="Schedule-Section">
        <h1 className="App-Section-Title">Schedule</h1>
        <Schedule/>
      </section>
      <hr />
      <section id="registration" className="Registration">
        <h1 className="App-Section-Title">Registration</h1>
      </section>
      <Registration/>
      <hr />
      <section id="inspiration" className="Inspiration">
        <h1 className="App-Section-Title">Inspiration</h1>
      </section>
      <hr />
      <Ideas />
      <hr />
      <section id="rules">
        <h1 className="App-Section-Title">Rules</h1>
        <Rules/>
      </section>
      <hr />
      <SVGSpriteSheet />
      <Footer/>
      <hr />
    </div >
  );
};

export default App;