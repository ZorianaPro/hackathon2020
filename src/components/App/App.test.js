import React from 'react';
import { shallow } from "enzyme";
import App from './';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';

describe('App', () => {
  const initialState = {
    appReducer: {}
  };
  const mockStore = configureStore();
  let store;

  beforeEach(() => {
    store = mockStore(initialState);
  });

  it('renders without crashing', () => {
    expect(shallow.bind(shallow,<Provider store={store}><App/></Provider>)).not.toThrow();
  });

  it('renders as expected', () => {
    expect(shallow(<Provider store={store}><App/></Provider>)).toMatchSnapshot();
  });
});

