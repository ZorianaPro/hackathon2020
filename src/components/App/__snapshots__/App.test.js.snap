// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders as expected 1`] = `
<div
  className="App"
>
  <Header />
  <section
    className="App-Main-Section"
    id="about"
  >
    <div
      className="App-Main-Section-Container"
    >
      <div
        className="App-Main-Section-Info"
      >
        <div
          className="App-Main-Section-Title"
        >
          Cool name for our cool hackathon
        </div>
        <div
          className="App-Main-Section-Date"
        >
          <DateSVG />
          <p>
             12-15 December 2020 
          </p>
        </div>
        <div
          className="App-Main-Section-Location"
        >
          <LocationSVG />
          <p>
             Online 
          </p>
        </div>
        <NavHashLink
          scroll={[Function]}
          smooth={true}
          to="/#registration"
        >
          <Button
            title="Registration"
          />
        </NavHashLink>
      </div>
    </div>
  </section>
  <hr />
  <section
    className="Schedule-Section"
    id="schedule"
  >
    <h1
      className="App-Section-Title"
    >
      Schedule
    </h1>
    <Schedule />
  </section>
  <hr />
  <Registration />
  <hr />
  <section
    className="Inspiration"
    id="inspiration"
  >
    <h1
      className="App-Section-Title"
    >
      Inspiration
    </h1>
  </section>
  <hr />
  <Ideas />
  <hr />
  <section
    id="rules"
  >
    <h1
      className="App-Section-Title"
    >
      Rules
    </h1>
    <Rules />
  </section>
  <hr />
  <SVGSpriteSheet />
  <Footer />
  <hr />
</div>
`;
